diff --git a/src/circllhist.c b/src/circllhist.c
index 8ff1221..466343b 100644
--- a/src/circllhist.c
+++ b/src/circllhist.c
@@ -69,7 +69,11 @@ static const hist_bucket_t hbnan = { (int8_t)0xff, 0 };
 } while(0)
 #define ASSERT_GOOD_BUCKET(hb) assert(hist_bucket_is_valid(hb))
 #else
+#if !defined(_MSC_VER)
 #define unlikely(x) __builtin_expect(!!(x), 0)
+#else
+#define unlikely(x) (x)
+#endif
 #define ASSERT_GOOD_HIST(h)
 #define ASSERT_GOOD_BUCKET(hb)
 #endif
